api/api/controllers/search_controller.py

```diff
@@ -300,8 +315,49 @@ def create_search_query(
     )
 
 
-def search(
-    search_params: media_serializers.MediaSearchRequestSerializer,
+def build_collection_query(
+    search_params: MediaListRequestSerializer,
+    collection_params: dict[str, str],
+):
+    """
+    Build the query to retrieve items in a collection.
+    :param collection_params: `tag`, `source` and/or `creator` values from the path.
+    :param search_params: the validated search parameters.
+    :return: the search client with the query applied.
+    """
+    search_query = {"filter": [], "must": [], "should": [], "must_not": []}
+    # Apply the term filters. Each tuple pairs a filter's parameter name in the API
+    # with its corresponding field in Elasticsearch. "None" means that the
+    # names are identical.
+    filters = [
+        # Collection filters allow a single value.
+        ("tag", "tags.name"),
+        ("source", None),
+        ("creator", None),
+    ]
+    for serializer_field, es_field in filters:
+        if serializer_field in collection_params:
+            if not (argument := collection_params.get(serializer_field)):
+                continue
+            parameter = es_field or serializer_field
+            search_query["filter"].append({"term": {parameter: argument}})
```

Comment:
Alternative that uses `match` for the tags:

```suggestion
    filters = [
        # Collection filters allow a single value.
        ("tag", "tags.name", "match"),
        ("source", None, "term"),
        ("creator", None, "term"),
    ]
    for serializer_field, es_field, query_type in filters:
        if serializer_field not in collection_params:
            continue
        if not (argument := collection_params.get(serializer_field)):
            continue

        parameter = es_field or serializer_field

        match query_type:
            case "match":
                query = {"match": {parameter: {"query": argument}}}
            case "term":
                query = {"term": {parameter: argument}}

        search_query["filter"].append(query)
```

If we take this approach, then we can cut down on the documentation complexity too, I guess, by just saying it uses regular fuzzy matching. But I know this is something we've talked about a lot, and I'm not totally remembering why we decided to use a terms query for tags, so forgive me if I'm just re-treading ground we've already covered.